tests/unit_tests/cuda/propagator_cuda_kernel.hpp

```diff
@@ -58,33 +56,53 @@ constexpr scalar pos_diff_tolerance = 1e-3;
 
 namespace detray {
 
-template <template <typename...> class vector_type>
-struct track_inspector {
+template <template <typename...> class vector_t>
+struct track_inspector : actor {
+
+    struct track_inspector_state {
+        using intersection_t = line_plane_intersection;
+
+        track_inspector_state(vecmem::memory_resource &resource)
+            : _intersections(&resource) {}
+
+        DETRAY_HOST_DEVICE
+        track_inspector_state(vector_t<intersection_t> intersection_record)
+            : _intersections(intersection_record) {}
 
-    track_inspector(vecmem::memory_resource& resource)
-        : _intersections(&resource) {}
+        vector_t<intersection_t> _intersections;
+    };
 
-    DETRAY_HOST_DEVICE
-    track_inspector(vector_type<intersection_t> intersection_record)
-        : _intersections(intersection_record) {}
+    using state_type = track_inspector_state;
+
+    template <typename propagator_state_t>
+    DETRAY_HOST_DEVICE void operator()(
+        state_type &inspector_state,
+        const propagator_state_t &prop_state) const {
+
+        const auto &navigation = prop_state._navigation;
 
-    template <typename navigator_state_t, typename stepper_state_t>
-    DETRAY_HOST_DEVICE void operator()(const navigator_state_t& navigation,
-                                       const stepper_state_t& /*stepping*/) {
         // Record when status == e_on_target
         if (navigation.status() == navigation::status::e_on_target) {
-            _intersections.push_back(*navigation.current());
+            inspector_state._intersections.push_back(*navigation.current());
         }
     }
-
-    vector_type<intersection_t> _intersections;
 };
 
+// Assemble propagator type
+using inspector_host_t = track_inspector<vecmem::vector>;
+using inspector_device_t = track_inspector<vecmem::device_vector>;
+using actor_chain_host_t = actor_chain<thrust::tuple, inspector_host_t>;
+using actor_chain_device_t = actor_chain<thrust::tuple, inspector_host_t>;
```

Comment:
Have you tried using `std::tuple` for this?

Reply:
Actually no. i was under the impression that ```thrust::tuple``` was also used host-side in the cuda tests. I can try ```std::tuple```, but that is basically the ```common/test/propagator.inl``` test, right? I think, it might be time to define proper headers in ```common/test/*```. What do you think?

Reply:
Nice, seems it actually runs with ```std::tuple```. Although at this point it does not make much of a difference for this PR anymore (except compilation time I guess). Regardless, I removed the template now :)

Reply:
Ah OK. It's good to confirm it is working with `std::tuple` 
Actually I think it is better to keep the tuple template for the moment. (as `std::tuple` might not work well without `--expt-relaxed-constexpr`)
Could you roll back one commit?

Reply:
Sure